common/djangoapps/util/philu_utils.py

```diff
@@ -17,3 +24,23 @@ def extract_utm_params(input_dict):
     ]
 
     return {key: value for key, value in input_dict.items() if key in utm_keys}
+
+
+@deconstructible
+class UploadToPathAndRename(object):
+    """
+    Rename file uploaded by user.
+    """
+
+    def __init__(self, path, field_name):
+        self.sub_path = path
+        self.field_name = field_name
+
+    def __call__(self, instance, filename):
+        file_extension = filename.split('.')[-1] if filename else ''
+
+        name_prefix = 'image' if isinstance(instance._meta.get_field(self.field_name), ImageField) else 'file'
```

Comment:
instead of doing all this you can simplify this by removing field_name and pass a flag 
`is_image=False`
in init of class

and in case of ImageField you can send this flag True, instead you don't even need to pass as default value is False

Reply:
This will benefit you to avoid accessing protected member of class that is **_meta**